### DISCUSSION TITLE
Increase maximum UDP response packet size?

### DISCUSSION THREAD

**Discussion by PeterDaveHello** [DNSCrypt/dnscrypt-proxy#2044] [Potential issues]
Using dnscrypt-proxy as upstream for Pi-Hole to help encrypt DNS traffic will receive this warning in some scenarios(not all of them, but I haven't confirmed the critical point that impact this issue):

> Warning in dnsmasq core:
> 
> `reducing DNS packet size for nameserver 127.0.0.1 to 1232`
> 
> Check out [our documentation](https://docs.pi-hole.net/ftldns/dnsmasq_warn/) for further information.

> ⚠️ **reducing DNS packet size for nameserver `ADDRESS` to `SAFE_PKTSZ`**
> 
> When receiving answers from upstream only with a smaller maximum DNS packet size, dnsmasq warns about this and remembers this decision per server for some time (defaulting to 60 seconds).
> 
> If you see this message continuously, you are affected by some unusual truncation on the path from your Pi-hole to the configured upstream server. You can get rid of the warning by adding a config file like `/etc/dnsmasq.d/99-edns.conf` and adding
> 
> ```
> edns-packet-max=1232
> ``` 
>
> After running `pihole restartdns` your Pi-hole will not even try larger packet sizes (the default is 4096). Check out our [unbound guide](https://docs.pi-hole.net/guides/dns/unbound/) for a comment about the particular value of `1232`.

Not sure if it's something worthy adjustment here? Many thanks!

**Comment by jedisct1** [comment]
`dnscrypt-proxy` can return responses up to 4096 bytes, depending on the EDNS value included in queries.

Are Pi-Hole and `dnscrypt-proxy` running on the same host?

**Reply by PeterDaveHello** [reply]
same host, but Pi-Hole is inside a Docker container, using https://github.com/pi-hole/docker-pi-hole.
